File: examples/cpp/Misc/save_load.cpp
#include <iostream>
#include <optional>
#include <utility>

#include "depthai/depthai.hpp"

int main() {
    dai::Pipeline pipeline;
    auto camera = pipeline.create<dai::node::Camera>()->build();
    auto frameQueue = camera->requestOutput(std::make_pair(640, 400))->createOutputQueue();

    auto depth = pipeline.create<dai::node::Depth>();
    depth->build(dai::node::Depth::Algorithm::AUTO, std::nullopt, std::make_pair(640u, 400u));
    auto pointCloud = pipeline.create<dai::node::PointCloud>();
    depth->depth().link(pointCloud->inputDepth);
    auto pointCloudQueue = pointCloud->outputPointCloud.createOutputQueue();

    pipeline.start();
    const auto capturedFrame = frameQueue->get<dai::ImgFrame>();
    const auto capturedPointCloud = pointCloudQueue->get<dai::PointCloudData>();
    if(!capturedFrame || !capturedPointCloud) {
        std::cerr << "Failed to capture a frame or point cloud\n";
        return 1;
    }
    capturedFrame->save("./frame.dai");
    capturedPointCloud->save("./pointcloud.dai");
    pipeline.stop();

    dai::ImgFrame frame;
    frame.load("./frame.dai");
    std::cout << "Loaded frame: " << frame.getWidth() << 'x' << frame.getHeight() << '\n';

    dai::PointCloudData loadedPointCloud;
    loadedPointCloud.load("./pointcloud.dai");
    std::cout << "Loaded point cloud: " << loadedPointCloud.getPoints().size() << " points\n";
}

File: examples/python/Misc/save_load.py
#!/usr/bin/env python3

import depthai as dai


with dai.Pipeline() as pipeline:
    camera = pipeline.create(dai.node.Camera).build()
    queue = camera.requestOutput((640, 400)).createOutputQueue()
    depth = pipeline.create(dai.node.Depth).build(dai.node.Depth.Algorithm.AUTO, None, (640, 400))
    point_cloud = pipeline.create(dai.node.PointCloud)
    depth.depth.link(point_cloud.inputDepth)
    point_cloud_queue = point_cloud.outputPointCloud.createOutputQueue()

    pipeline.start()
    queue.get().save("./frame.dai")
    point_cloud_queue.get().save("./pointcloud.dai")

frame = dai.ImgFrame()
frame.load("./frame.dai")
print(f"Loaded frame: {frame.getWidth()}x{frame.getHeight()}")

point_cloud = dai.PointCloudData()
point_cloud.load("./pointcloud.dai")
print(f"Loaded point cloud: {len(point_cloud.getPoints())} points")
